tests/test_cli.py

```diff
@@ -73,3 +73,56 @@ def _capture_match_gpus_spec(file_path, gpus: str):
         _capture_match_gpus_spec(f.name, 'v100:1')
         _capture_match_gpus_spec(f.name, 'V100:0.5')
         _capture_match_gpus_spec(f.name, 'V100')
+
+
+def test_show_gpus_gpu_name():
+    """
+    This is a conprehensive test quite for `sky show-gpus`.
+    The tests below correspond to the following terminal commands,
+    in order:
+
+    -> sky show-gpus
+    -> sky show-gpus --all
+    -> sky show-gpus V100:4
+    -> sky show-gpus :4
+    -> sky show-gpus V100:0
+    -> sky show-gpus V100:-2
+    -> sky show-gpus --cloud lambda
+    -> sky show-gpus --cloud lambda --all
+    -> sky show-gpus V100:4 --cloud lambda
+    -> sky show-gpus V100:4 --cloud lambda --all
+    """
+    cli_runner = cli_testing.CliRunner()
+    result = cli_runner.invoke(cli.show_gpus, [])
+    assert not result.exit_code
+
+    result = cli_runner.invoke(cli.show_gpus, ['--all'])
+    assert not result.exit_code
+
+    result = cli_runner.invoke(cli.show_gpus, ['V100:4'])
+    assert not result.exit_code
+
+    result = cli_runner.invoke(cli.show_gpus, [':4'])
+    assert not result.exit_code
+
+    result = cli_runner.invoke(cli.show_gpus, ['V100:0'])
+    assert isinstance(result.exception, SystemExit)
+
+    result = cli_runner.invoke(cli.show_gpus, ['V100:-2'])
+    assert isinstance(result.exception, SystemExit)
+
+    result = cli_runner.invoke(cli.show_gpus, ['--cloud', 'lambda'])
+    assert not result.exit_code
+
+    result = cli_runner.invoke(cli.show_gpus, ['--cloud', 'lambda', '--all'])
```

Comment:
Can we also add a test to exercise the `--region` flag?

Reply:
Just added a single test since the only way I found specifying the regions was through hardcoding it in specific to the cloud provider specified